### ISSUE TITLE
resolved hangs when result is cached

### ISSUE THREAD

**Issue by hvenev** [systemd/systemd#2145]
Lookup example.org. Result OK.
Lookup example.com. Result OK.
Lookup example.org. Hang. Ctrl-C.
Lookup example.org. Resolved restarts (Code should not be reached 'Transaction not started?' at src/resolve/resolved-dns-query.c:201). Result OK.
Lookup example.org. Hang. Ctrl-C.
Lookup example.org. Resolved restarts. Result OK.

I haven't tried waiting for the cached data to expire.

**Comment by poettering** [comment]
which version of resovled is this?

**Comment by hvenev** [comment]
master 4 days ago, I think a0361331758dbdb2375d6f871bc959116b699e31.
cbdb172efc76870dbecffebbd4c180810e72e8e4 is also broken.

```
Dec 15 12:21:20 hristo.venev.name systemd-resolved[22456]: Using system hostname 'hristo'.
Dec 15 12:21:20 hristo.venev.name systemd[1]: Started Network Name Resolution.
Dec 15 12:21:24 hristo.venev.name systemd-resolved[22456]: Switching to system DNS server 8.8.8.8.
Dec 15 12:21:24 hristo.venev.name systemd-resolved[22456]: Switching to DNS server 10.111.0.1 for interface br-lan.
Dec 15 12:21:50 hristo.venev.name systemd-resolved[22456]: Code should not be reached 'Transaction not started?' at /var/tmp/portage/sys-apps/systemd-9999/work/systemd-9999/src/resolve/resolved-dns-query.c:201, 
function dns_query_candidate_state(). Aborting.
Dec 15 12:21:50 hristo.venev.name systemd[1]: systemd-resolved.service: Main process exited, code=killed, status=6/ABRT
Dec 15 12:21:50 hristo.venev.name systemd[1]: systemd-resolved.service: Unit entered failed state.
Dec 15 12:21:50 hristo.venev.name systemd[1]: systemd-resolved.service: Failed with result 'signal'.
Dec 15 12:21:50 hristo.venev.name systemd[1]: systemd-resolved.service: Service has no hold-off time, scheduling restart.
Dec 15 12:21:50 hristo.venev.name systemd[1]: Stopped Network Name Resolution.
Dec 15 12:21:50 hristo.venev.name systemd[1]: Starting Network Name Resolution...
Dec 15 12:21:50 hristo.venev.name systemd-resolved[22463]: Using system hostname 'hristo'.
Dec 15 12:21:50 hristo.venev.name systemd[1]: Started Network Name Resolution.
Dec 15 12:21:51 hristo.venev.name systemd-resolved[22463]: Switching to system DNS server 8.8.8.8.
Dec 15 12:21:51 hristo.venev.name systemd-resolved[22463]: Switching to DNS server 10.111.0.1 for interface br-lan.
```

**Comment by poettering** [comment]
I am pretty sure this is fixed with today's git version! Please verify!
